### PR TITLE
feat: mdx compilation in rust

### PR DESCRIPTION
**Pull request by sanyuan0704**
## Description

1. 开发 mdx-rs-loader，接入 @modern-js/mdx-rs-binding，增加 experimentalMdxRs 选项进行开启
2. 使用 highlight.js 进行代码块运行时着色，优化编译性能
3. 修复一些文档死链

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/main-doc/modern.config.ts [hunk 1]
```diff
@@ -133,7 +134,7 @@ export default defineConfig({
         dataUriLimit: 0,
       },
       dev: {
-        startUrl: true,
+        startUrl: false,
```
Comment (chenjiahan) [inline packages/document/main-doc/modern.config.ts:137]: 为啥要禁用 startUrl 嘞

File: packages/document/builder-doc/docs/en/config/source/transformImport.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ Array<{
 
 - **Default:**
 
-When the [antd component library](https://www.npmjs.com/package/antd) <= 4.x version is installed in the project, Builder will automatically add the corresponding on-demand import configuration. The default configuration is as follows:
+When the [antd component library](https://www.npmjs.com/package/antd) &lt;= 4.x version is installed in the project, Builder will automatically add the corresponding on-demand import configuration. The default configuration is as follows:
```
Comment (chenjiahan) [inline packages/document/builder-doc/docs/en/config/source/transformImport.md:22]: <= 不能用了嘛

File: packages/cli/doc-core/package.json [hunk 3]
```diff
@@ -136,7 +138,8 @@
     "rimraf": "^3.0.2",
     "typescript": "^4.8.4",
     "vite": "^4.0.4",
-    "vitest": "0.21.1"
+    "vitest": "0.21.1",
+    "@rspack/core": "^0.1.1"
```
Comment (chenjiahan) [inline packages/cli/doc-core/package.json:142]: how about 0.1.4

File: packages/document/main-doc/modern.config.ts (hunk 1 above)
Comment (sanyuan0704) [reply packages/document/main-doc/modern.config.ts:137]: 每次更新配置打开新的页面体验不太好

**Review by chenjiahan — DISMISSED** [review]

**Pull request merged**